### PR TITLE
Don't log passwords

### PR DESCRIPTION
**Pull request by sethvargo**

### CODE REVIEW

**Comment by sethvargo** [comment]
/assign @mikehelmick 

This still isn't "perfect" as the password would be printed if someone called `Printf` with "%v" or "%#v", but it's a good start.

File: internal/setup/setup.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func SetupWith(ctx context.Context, config interface{}, l envconfig.Lookuper) (*
 		// Update serverEnv setup.
 		serverEnvOpts = append(serverEnvOpts, serverenv.WithSecretManager(sm))
 
-		logger.Infof("secret manager configuration: %+v", smConfig)
+		logger.Infow("secret manager", "config", smConfig)
```
Comment (sethvargo) [inline internal/setup/setup.go:123]: This gives us structured logging, yay

**Comment by crwilcox** [comment]
lgtm, but will let @mikehelmick approve :)

File: internal/database/config.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ type Config struct {
 	Port               string        `env:"DB_PORT, default=5432"`
 	SSLMode            string        `env:"DB_SSLMODE, default=require"`
 	ConnectionTimeout  int           `env:"DB_CONNECT_TIMEOUT"`
-	Password           string        `env:"DB_PASSWORD"`
+	Password           string        `env:"DB_PASSWORD" json:"-"`
```
Comment (mikehelmick) [inline internal/database/config.go:28]: so, the json:"-" is interpreted by the logging infrastructure?

Comment (mikehelmick) [reply internal/database/config.go:28]: it's fine if so - but would deserving of a comment and maybe a link

Comment (sethvargo) [reply internal/database/config.go]: Added comment.

**Comment by sethvargo** [comment]
@mikehelmick re-review plz

**Pull request merged**